src/nat/middleware/dynamic/dynamic_function_middleware.py

```
@@ -0,0 +1,959 @@
+
+        discovered_component = DiscoveredComponent(name=llm_name,
+                                                   component_type=ComponentGroup.LLMS,
+                                                   instance=llm_client,
+                                                   config=None,
+                                                   callable_functions=all_functions)
+        self._workflow_inventory.llms.append(discovered_component)
+
+        for function_name in all_functions:
+            try:
+                self._register_component_function(discovered_component, function_name)
+            except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)